tools/verdict-back-ends/verdict-bundle/verdict-lustre-translator/src/main/java/edu/uiowa/clc/verdict/lustre/VDMLustre2Kind2.java

```diff
@@ -0,0 +1,583 @@
+            pb.addContract(visit(modelContract));
+        }
+
+        for (Node node : modelProgram.getNodeDeclaration()) {
+            if (node.isIsFunction() != null && node.isIsFunction()) {
+                if (node.isIsImported() != null && node.isIsImported()) {
+                    pb.importFunction(visitImportedComponent(node));
+                } else {
+                    if (node.getBody().isIsMain() != null && node.getBody().isIsMain()) {
+                        pb.setMain(node.getName());
+                    }
+                    pb.addFunction(visitComponent(node));
+                }
+            } else {
+                if (node.isIsImported() != null && node.isIsImported()) {
+                    pb.importNode(visitImportedComponent(node));
+                } else {
+                    if (node.getBody().isIsMain() != null && node.getBody().isIsMain()) {
+                        pb.setMain(node.getName());
+                    }
+                    pb.addNode(visitComponent(node));
+                }
+            }
+        }
+
+        return pb;
+    }
+
+    /**
+     * Translate VDM model datatype to Kind 2 Lustre type.
+     *
+     * @param modelType the VDM model type
+     * @return the Kind 2 Lustre type
+     */
+    private static Type visit(DataType modelType) {
+        if (modelType.getPlainType() != null) {
+            return TypeUtil.named(modelType.getPlainType().value());
+        }
+
+        if (modelType.getUserDefinedType() != null) {
+            return TypeUtil.named(modelType.getUserDefinedType());
+        }
+
+        if (modelType.getSubrangeType() != null) {
+            SubrangeType subrange = modelType.getSubrangeType();
+            return TypeUtil.intSubrange(subrange.getLowerBound(), subrange.getUpperBound());
+        }
+
+        if (modelType.getArrayType() != null) {
+            return TypeUtil.array(
+                    visit(modelType.getArrayType().getDataType()),
+                    Integer.parseInt(modelType.getArrayType().getDimension()));
+        }
+
+        if (modelType.getTupleType() != null) {
+            return TypeUtil.tuple(
+                    modelType.getTupleType().getDataType().stream()
+                            .map(dt -> visit(dt))
+                            .collect(Collectors.toList()));
+        }
+
+        if (modelType.getEnumType() != null) {
+            return TypeUtil.enumeration(modelType.getEnumType().getEnumValue());
+        }
+
+        if (modelType.getRecordType() != null) {
+            // fields in the map have the same order as the RecordFields in the RecordField
+            // list
+            Map<String, Type> fields = new TreeMap<>((x, y) -> -1);
+            for (RecordField field : modelType.getRecordType().getRecordField()) {
+                fields.put(field.getName(), visit(field.getType()));
+            }
+            return TypeUtil.record(fields);
+        }
+
+        throw new IllegalArgumentException(
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Use of identifier that is a keyword in later versions of Java (NM_FUTURE_KEYWORD_USED_AS_IDENTIFIER)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-use-of-identifier-that-is-a-keyword-in-later-versions-of-java-nm-future-keyword-used-as-identifier).
The identifier is a word that is reserved as a keyword in later versions of Java, and your code will need to be changed in order to compile it in later versions of Java.